Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -165,6 +165,7 @@ final class PurchasesOrchestrator {
         }
 
         Task {
+            await diagnosticsSynchronizer?.clearDiagnosticsFileIfTooBig()
             await syncDiagnosticsIfNeeded()
```

Comment:
When we do actually clear diagnostics if the file is too large, we'll only have one event - the "file cleared event". 
It's kind of a bummer because we're losing a bunch of events unnecessarily. 

I was thinking that we could optimize this a bit by doing the "clear if too big" part at event generation time instead of at syncing time. This would cut down on the events thrown away. 

This would optimize cases where for example you: 
before:
- have max file size 10
- have 15 events
- sync, now you delete 15 events and generate the "file nuked" event and sync that

after: 
- have max file size 10
- when you try to add event 11, you see that you've gone over and kill the previous 10
- add the next 4
- sync, now you have 5 events that you can sync

bonus points: file size is guaranteed to never go over the limit, regardless of any other logic.

Reply:
That is a good point. Will update android to this new logic and update this PR as well. Thanks!